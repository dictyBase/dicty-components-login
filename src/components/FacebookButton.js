// @flow
import React from "react"
import { Flex, Box } from "rebass"
import FontAwesomeIcon from "@fortawesome/react-fontawesome"
import faFacebookF from "@fortawesome/fontawesome-free-brands/faFacebookF"
import {
  FontAwesomeIconContainer,
  FacebookButtonStyle
} from "../styles/Buttons"

/**
 * This is the Facebook social button. It has a FontAwesome icon and then some sign in text.
 */

const FacebookButton = () => {
  return (
    <Flex justify="center">
      <Box w={"90%"} mb={"5px"}>
        <FacebookButtonStyle size="large" variant="raised">
          <FontAwesomeIconContainer>
            <FontAwesomeIcon icon={faFacebookF} />
          </FontAwesomeIconContainer>&nbsp; Sign in with Facebook
        </FacebookButtonStyle>
      </Box>
    </Flex>
  )
}

export default FacebookButton
