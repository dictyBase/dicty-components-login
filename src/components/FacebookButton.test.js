import React from "react"
import renderer from "react-test-renderer"
import FacebookButton from "./FacebookButton"
import "jest-styled-components"

test("matching a snapshot of FacebookButton", () => {
  const component = renderer.create(<FacebookButton />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
