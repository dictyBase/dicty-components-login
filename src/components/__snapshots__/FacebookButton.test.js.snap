// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of FacebookButton 1`] = `
.c1 {
  box-sizing: border-box;
}

.c2 {
  box-sizing: border-box;
  width: 90%;
  margin-bottom: 5px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  border-right: 1px solid black;
  line-height: 45px;
  width: 45px;
  font-size: 1.8em;
}

.c3 {
  display: inline-block;
  padding: 6px 12px;
  margin-bottom: 0;
  font-size: 14px;
  font-weight: normal;
  line-height: 1.42857143;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -ms-touch-action: manipulation;
  touch-action: manipulation;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-image: none;
  border: 1px solid transparent;
  border-radius: 4px;
  position: relative;
  padding-left: 21px;
  text-align: left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  margin-top: 5px;
  color: #fff;
  border-color: rgba(0,0,0,0.2);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 10px 16px;
  font-size: 18px;
  line-height: 1.3333333;
  border-radius: 6px;
  background-color: #3b5998;
}

.c3.block {
  display: block;
  width: 100%;
}

.c3.large {
  padding: 10px 16px;
  font-size: 18px;
  line-height: 1.3333333;
  border-radius: 6px;
}

.c3.xs {
  padding: 1px 5px;
  font-size: 12px;
  line-height: 1.5;
  border-radius: 3px;
  width: 20%;
}

.c3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:focus,
.c3:hover,
.c3:active {
  color: #fff;
  background-color: #2d4373;
  border-color: rgba(0,0,0,0.2);
}

<div
  className="c0 c1"
  is={null}
>
  <div
    className="c2"
    is={null}
  >
    <a
      className="c3"
    >
      <i
        className="c4"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-facebook-f fa-w-9 "
          data-icon="facebook-f"
          data-prefix="fab"
          role="img"
          style={Object {}}
          viewBox="0 0 264 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M76.7 512V283H0v-91h76.7v-71.7C76.7 42.4 124.3 0 193.8 0c33.3 0 61.9 2.5 70.2 3.6V85h-48.2c-37.8 0-45.1 18-45.1 44.3V192H256l-11.7 91h-73.6v229"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </i>
      Â  Sign in with Facebook
    </a>
  </div>
</div>
`;
